lang/cs/Org.Apache.REEF.Common/Poison/PoisonedEventHandler.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Attributes;
+using Org.Apache.REEF.Utilities.Logging;
+using Org.Apache.REEF.Wake.Time.Event;
+using Org.Apache.REEF.Wake.Time.Runtime;
+using System;
+using System.Reactive;
+using System.Threading.Tasks;
+
+namespace Org.Apache.REEF.Common.Poison
+{
+    /// <summary>
+    /// Handler to process an event in a way that has certain probability of failure within certain inverval of time.
+    /// </summary>
+    /// <typeparam name="T">The type of event</typeparam>
+    [Private]
+    public class PoisonedEventHandler<T> : IObserver<T>
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(PoisonedEventHandler<T>));
+
+        private readonly double _crashProbability;
+        private readonly int _crashTimeout;
+        private readonly int _crashMinDelay;
+        private readonly RuntimeClock _clock;
+
+        private readonly Random rand = new Random();
+
+        [Inject]
+        private PoisonedEventHandler(
+            [Parameter(typeof(CrashProbability))] double crashProbability,
+            [Parameter(typeof(CrashTimeout))] int crashTimeout,
+            [Parameter(typeof(CrashMinDelay))] int crashMinDelay,
+            RuntimeClock clock) 
+        {
+            _crashProbability = crashProbability;
+            _crashTimeout = crashTimeout;
+            _crashMinDelay = crashMinDelay;
+            _clock = clock;
+        }
+
+        public void OnNext(T value)
```

Comment:
Add documentation here. It would be good to say explicitly in words the behavior here.